### PR TITLE
Ianhelle/mordor provider 2020 09 29

### PR DESCRIPTION
**Pull request by ianhelle**
Mordor data provider and browser

### CODE REVIEW

File: docs/source/data_acquisition/MordorData.rst [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+Searching for Queries with QueryProvider.search_queries()
+---------------------------------------------------------
+
+You can use the provider
+:py:func:`search_queries<msticpy.data.drivers.mordor_driver.MordorDriver.search_queries>`
+function to search for queries for matching required attributes.
+
+This function takes a single string parameter - ``search``.
+
+The search text treated as a single text string to search for. It also
```
Comment (petebryan) [inline docs/source/data_acquisition/MordorData.rst:170]: Don't really understand this line.

File: docs/source/data_acquisition/MordorData.rst [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+
+The top scrollable list is a list of the Mordor datasets. Selecting one
+of these updates the data in the lower half of the browser.
+
+Filter Drop-down
+----------------
+
+To narrow your search you can filter using a text search or filter by
+Mitre Attack Techniques or Tactics. Click on the arrow to open the
```
Comment (petebryan) [inline docs/source/data_acquisition/MordorData.rst:264]: ATT&CK

Comment (petebryan) [reply docs/source/data_acquisition/MordorData.rst:264]: Also maybe have a link to ATT&CK here

File: msticpy/data/drivers/mordor_driver.py [hunk 3]
```diff
@@ -0,0 +1,809 @@
...
+    """
+    if not silent:
+        print("Extracting", file_name)
+
+    file_path = Path(save_folder).joinpath(file_name)
+    if not use_cached or not file_path.is_file():
+        zip_file.extract(file_name, path=save_folder)
+
+    out_df = pd.DataFrame()
+    if file_path.suffix.lower() == ".json":
```
Comment (petebryan) [inline msticpy/data/drivers/mordor_driver.py:705]: would it not be more efficent just to use pandas read_json() here?
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json.html?highlight=read_json

I tested this with a few of the morder datasets and pd.read_csv('file', line=True) seemed to work fine.

**Review by petebryan — COMMENTED** [review]
Some comments but otherwise looks good.

**Review by petebryan — APPROVED** [review]

File: msticpy/data/drivers/mordor_driver.py (hunk 3 above)
Comment (ianhelle) [reply msticpy/data/drivers/mordor_driver.py:705]: Yep, you're right. I hadn't seen the lines parameter and couldn't get it to work.

File: docs/source/data_acquisition/MordorData.rst (hunk 1 above)
Comment (ianhelle) [reply docs/source/data_acquisition/MordorData.rst]: Tried to re-write

File: docs/source/data_acquisition/MordorData.rst (hunk 2 above)
Comment (ianhelle) [reply docs/source/data_acquisition/MordorData.rst]: done

**Review by petebryan — APPROVED** [review]

File: docs/source/data_acquisition/MordorData.rst [hunk 4]
```diff
@@ -0,0 +1,396 @@
+Mordor data provider and browser
+================================
+
+This notebook provides a guided example of using the Mordor data
```
[the remark this answers is not in the document]
Comment (ianhelle) [reply docs/source/data_acquisition/MordorData.rst:4]: Good spot - I should be more careful generating docs from notebooks.

**Pull request merged**